tests/integration/test_action.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration test for actions."""
+
+import pytest
+from juju.application import Application
+
+from .conftest import TEST_EXTERNAL_HOSTNAME_CONFIG
+
+
+@pytest.mark.abort_on_fail
+async def test_get_certificate_action(
+    configured_application_with_tls: Application,
+):
+    """
+    arrange: Deploy the charm with valid config and tls integration.
+    act: Run the get-certificate action and run a sh command to check
+    the cert location on the unit.
+    assert: The output of both operations are valid.
+    """
+    action = await configured_application_with_tls.units[0].run_action(
+        "get-certificate", hostname=TEST_EXTERNAL_HOSTNAME_CONFIG
+    )
+    await action.wait()
+    assert "-----BEGIN CERTIFICATE-----" in action.results.get("certificate")
+
+    with pytest.raises(Exception):
```

Comment:
I didnt follow this one...Why rerunning the action but this time with pytest.raises(Exception)?

Reply:
Sorry I wasn't being clear here. Maybe I needed to add a comment explaining it. Basically the second time we run the action without the `hostname` parameter, and if the event fails pytest should raise an `Exception`

Reply:
Maybe the second running can be done by parameterizing the test? This way it would be more clear which test fails.

Reply:
Since the logic differs quite a bit, I moved the error case to a second test